### ISSUE TITLE
Add proguard configuration

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-directions-android#12]
When you build a client app with proguard enabled. 

I'm seeing following warnings that will stop the build from finishing:

![screen shot 2016-01-13 at 12 20 05](https://cloud.githubusercontent.com/assets/2151639/12292902/6a839040-b9f0-11e5-8f4e-fc524374807d.png)

I added following config to that client app to resolve this:
```
-dontwarn retrofit.**
-keep class retrofit.** { *; }
-keepattributes Signature
-keepattributes Exceptions
```

This problem can be handled at SDK level by introducing same configuration as in https://github.com/mapbox/mapbox-gl-native

**Comment by tobrun** [comment]
I just checked to see which version of retrofit this lib is using, 
which is retrofit2, normally the config should be:
```
-dontwarn retrofit2.**
-keep class retrofit2.** { *; }
-keepattributes Signature
-keepattributes Exceptions
```

But this is not working in my client app. 

@zugaldia 
did you recently upgrade from v1 to v2?

I'm using:
```
 compile ('com.mapbox.mapboxsdk:mapbox-android-directions:1.0.0@aar'){
```

**Comment by zugaldia** [comment]
Good catch. We're currently using `compile 'com.squareup.retrofit:retrofit:2.0.0-beta2'`. When we switch to `beta3` (any advantages?) then we'll be using the recent `retrofit2` namespace (`'com.squareup.retrofit2:retrofit:2.0.0-beta3'`). Let me add the rules in any case.

**Comment by tobrun** [comment]
I would wait for a stable release of retrofit with OkHttp 3.0.0, [now](https://twitter.com/JakeWharton/status/687536361952260096) in SNAPSHOT.

**Comment by zugaldia** [comment]
@tobrun good to know, thanks!
